components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -2481,31 +2481,7 @@ private int getIdentityProviderIdentifier(Connection dbConnection, String idPNam
         }
     }
 
-    private int getAuthenticatorIdentifier(Connection dbConnection, int idPId, String authnType)
-            throws SQLException, IdentityProviderManagementException {
-
-        String sqlStmt = null;
-        PreparedStatement prepStmt = null;
-        ResultSet rs = null;
-        try {
-            sqlStmt = IdPManagementConstants.SQLQueries.GET_IDP_AUTH_SQL;
-            prepStmt = dbConnection.prepareStatement(sqlStmt);
-            prepStmt.setInt(1, idPId);
-            prepStmt.setString(2, authnType);
-            rs = prepStmt.executeQuery();
-            if (rs.next()) {
-                return rs.getInt("ID");
-            } else {
-                throw new IdentityProviderManagementException("Cannot find authenticator : "
-                        + authnType);
-            }
-        } finally {
-            IdentityApplicationManagementUtil.closeResultSet(rs);
-            IdentityApplicationManagementUtil.closeStatement(prepStmt);
-        }
-    }
-
-    public boolean isSimilarIdPEntityIdsAvailble(String idPEntityId, int tenantId)
+    public boolean isSimilarIdPEntityIdsAvailble(String authenticatorName, String idPEntityId, int tenantId)
```

Comment:
Shall we make this method more generic ? We can make this method like isIdPAvailableForAuthenticatorProperty(String authenticatorName, String propertyName, String propertyValue, int tenantId). This way we can make it more usable.